### ISSUE TITLE
BUG/How-To: Use helmsman with only RBAC for a given namespace

### ISSUE THREAD

**Issue by Yanson** [Praqma/helmsman#53]
I want a desired state file for each namespace, allowing each team to own a namespace and run `helmsman` for it.

Given a service account `dev` which is a `ClusterAdmin` for a given namespace, helmsman returns the following error:

```yaml
namespaces:
  custom-ns:
    protected: false
    installTiller: true
    tillerServiceAccount: tiller
```
> 2018/08/06 12:00:27 ERROR: while deploying Helm Tiller in namespace [kube-system]: Error: error installing: deployments.extensions is forbidden: User "system:serviceaccount:custom-ns:dev" cannot create deployments.extensions in the namespace "kube-system": Unknown user "system:serviceaccount:custom-ns:dev"

If I add the `kube-system` details to my desired state file, I get the following error instead:

```yaml
namespaces:
  kube-system:
    protected: false
    installTiller: true # or false
    tillerServiceAccount: tiller
```

> 2018/08/06 12:04:46 ERROR: while deploying Helm Tiller in namespace [kube-system]: Error from server (Forbidden): serviceaccounts "tiller" is forbidden: User "system:serviceaccount:custom-ns:dev" cannot get serviceaccounts in the namespace "kube-system": Unknown user "system:serviceaccount:custom-ns:dev"

Pre-installing the `tiller` service account in the custom namespace is not a problem, but why does helmsman need to connect to `kube-system` and how/can I operate it without?

**Comment by Yanson** [comment]
I have created the tiller service account in the namespace before all these tests. It works fine with an account with full cluster admin, but does not work with a service account that only has cluster admin on a specific namespace.

I don't want other tenants to interact with `kube-system` - are you saying this is not possible right now?

Adding the service account to `settings` has the same effect as listing it in `name-spaces.kube-system.tillerServiceAccount` as far as I can see (though it says error validating rather than deploying Helm Tiller).

> 2018/08/07 14:41:30 ERROR: while validating service account: Error from server (Forbidden): serviceaccounts "tiller" is forbidden: User "system:serviceaccount:customer-marketing:dev" cannot get serviceaccounts in the namespace "kube-system": Unknown user "system:serviceaccount:customer-marketing:dev"

**Comment by Yanson** [comment]
@sami-alajrami Any further updates on this?

**Comment by sami-alajrami** [comment]
@Yanson sorry, I have missed this .. v1.4.0-rc fixes the creation of service accounts (if they don't already exist), however, it still deploys Tiller in `kube-system` and tries to create a service account and binds it to the clusteradmin role for that Tiller.  

So in your case, each team's desired state will try to deploy/upgrade Tiller in kube-system and if they don't have sufficient permissions on the cluster, this will fail. 

I will create a fix for this where you can specify that you don't want Tiller to be deployed in `kube-system` .. something like:

```yaml
namespaces:
  kube-system:
    installTiller: false
```

where the default behaviour remains to deploy Tiller in kube-system unless you set the ìnstallTiller: false` option for kube-system. 

I will update you again on the fix implementaion in the next couple of days, hopefully!

**Comment by sami-alajrami** [comment]
this is pushed into master. Still need documentation. Will be shipped in v1.4.0

**Comment by sami-alajrami** [comment]
this is shipped in v1.5.0 and documentation is updated in https://github.com/Praqma/helmsman/blob/master/docs/how_to/define_namespaces.md#preventing-tiller-deployment-in-kube-system
Closing this, but feel free to reopen if this still does not meet your needs.

**Comment by Yanson** [comment]
Hi, @sami-alajrami thanks for the changes. With 1.5 I am seeing this error:
```
2018/09/07 14:43:10 INFO: listing all existing releases in namespace [ custom-namespace ]...
2018/09/07 14:43:11 INFO: checking if any Helmsman managed releases are no longer tracked by your desired state ...
2018/09/07 14:43:11 INFO: getting helm releases which are managed by Helmsman.
2018/09/07 14:43:11 Error from server (Forbidden): configmaps is forbidden: User "system:serviceaccount:custom-namespace:dev" cannot list configmaps at the cluster scope: Unknown user "system:serviceaccount:custom-namespace:dev"
```
